metagraph/api/python/metagraph/client.py

```diff
@@ -48,22 +48,33 @@ def search(self, sequence: Union[str, Iterable[str]],
                 f"discovery_threshold should be between 0 and 1 inclusive. Got {discovery_threshold}")
 
         if align:
-            json_obj = self.align(sequence, **align_params)
+            alignments = self.align(sequence, **align_params)
 
             def to_fasta(df):
                 fasta = []
                 for i in range(df.shape[0]):
                     fasta.append(f">{df.loc[i, 'seq_description']}\n{df.loc[i, 'sequence']}")
                 return '\n'.join(fasta)
 
-            sequence = to_fasta(helpers.df_from_align_result(json_obj))
+            sequence = to_fasta(helpers.df_from_align_result(alignments))
 
         param_dict = {"count_labels": True,
                       "discovery_fraction": discovery_threshold,
                       "num_labels": top_labels,
                       "with_signature": with_signature}
 
-        return self._json_seq_query(sequence, param_dict, "search")
+        search_results = self._json_seq_query(sequence, param_dict, "search")
+
+        if align:
+            if len(alignments) == len(search_results):
+                # Zip best alignment results
+                for alignment, search_result in zip(alignments, search_results):
+                    if 'alignments' in alignment and len(alignment['alignments']) > 0:
+                        search_result['best_alignment'] = alignment['alignments'][0]
+                    else:
+                        search_result['best_alignment'] = {}
+
```

Comment:
Could you clarify what is going on here?

Why is this only for `len(alignments) == len(search_results)`? Can they have different length?

Reply:
With search with alignment it should only retrieve the best alignment I believe? Although I agree now I will adjust this to make it more robust.

Reply:
If it must always return 1 alignment, this should be an assertion instead of a check

Reply:
ping